name.abuchen.portfolio/src/name/abuchen/portfolio/model/Bookmark.java

```diff
@@ -42,11 +48,63 @@ public boolean isSeparator()
 
     public String constructURL(Security security)
     {
-        String url = pattern.replace("{tickerSymbol}", encode(security.getTickerSymbol())); //$NON-NLS-1$
-        url = url.replace("{isin}", encode(security.getIsin())); //$NON-NLS-1$
-        url = url.replace("{wkn}", encode(security.getWkn())); //$NON-NLS-1$
-        url = url.replace("{name}", encode(security.getName())); //$NON-NLS-1$
+        boolean replacementDone = Boolean.FALSE;
+        String url = pattern;
+        Class<? extends Security> securityClass = security.getClass();
+        HashMap<String, String> types = new HashMap<String, String>();
+        TreeMap <Integer,Entry<String, String>> replaceTypes = new TreeMap<Integer,Entry<String, String>>();
+
+        
+        
+        types.put("{tickerSymbol}", "getTickerSymbol");
```

Comment:
Warum keine Lambda Function? Per Reflection geht auch, aber man bekommt keine Compilefehler mehr, kann nicht in der IDE umbenennt.

Sowas wie
```
        Map<String, Function<Security, String>> types = new HashMap<>();
        types.put("isin", Security::getIsin);
```